### PR TITLE
Show Network locations and shares and add option to map network drives

### LINKED ISSUE #1523: Ability to map network drives

### LINKED ISSUE #2932: Add Network Path
**Network Path**
-Add the Network Directory so that we can access files over the Local Network

**Context Menu**
-Add _Close other tabs_ when right-clicking a tab, just like Chrome's tab management

### LINKED ISSUE #3488: Crash when viewing properties of hidden/system file
When I view the detail of the C:\Users\Administrator\Music\desktop.ini, my app crashed.
these are my debug.txt and debug_fulltrust.txt
[debug.txt](https://github.com/files-community/Files/files/5941771/debug.txt)
[debug_fulltrust.txt](https://github.com/files-community/Files/files/5941772/debug_fulltrust.txt)

### LINKED ISSUE #3551: Extracting a file using WinRar the app closed suddenly
**Describe the bug**
I have a file in .zip, so I have installed it WinRar 5.70, trying to unzip the file the app closed suddenly.
[debug_fulltrust.txt](https://github.com/files-community/Files/files/5973123/debug_fulltrust.txt)

**To Reproduce**
1. Select the file.
2. Right Click in the file.
3. Select the option extract to
4. App closed

**Expected behavior**
Do not close the app.

**Screenshots**
NA

**Desktop (please complete the following information):**
 - OS Version: Windows 10 19042.746
 - App version: 1.2.2.0

**Additional context**
When you open the app the file was extracted successfully, so I delete the extracted file, and I did it again and no issue.

**Log file**
Attached

### PR DESCRIPTION
**Pull request by gave92**
Fixes #1523
Fixes #2932
Fixes #3488
Fixes #3551

Add support for networks shortcuts and places.

Todo:
- [x] Show icons in ftp shares
- [x] Add ability to add new mapped drives &rarr; dialog opens behind Files window, solution: make our own dialog
- [x] ~No possibility to "cancel" loading network places folder~ &rarr; next time
- [x] ~Mapped network drives are still shown under "Drives"~ &rarr; I don't want to duplicate the work done [here](https://github.com/files-community/Files/tree/sidebar-overhaul)
- [x] ~App freezes for a few seconds when clicking a network drive on the sidebar (reading folder layout)~ &rarr; only when the drive is disconnected
- [x] ~Fulltrust process doesn't close if disconnected network drives are present~ &rarr; pre-existing issue

Files
image

Explorer
image

### CODE REVIEW

**Comment by gave92** [comment]
Hi @yair100, did you get any chance to test this?

**Comment by yaichenbaum** [comment]
@gave92 I don't have any network to test with right now.

**Comment by gave92** [comment]
@yair100 no problem. As I don't have it either, to test this I've installed 2/3 apps on my android phone to simulate network shares xD ("Server WebDav", "ES File Explorer", "LAN drive")

**Comment by yair100** [comment]
> @yair100 no problem. As I don't have it either, to test this I've installed 2/3 apps on my android phone to simulate network shares xD ("Server WebDav", "ES File Explorer", "LAN drive")

Good idea! By the way, how can I map a new drive from Files?

**Comment by gave92** [comment]
@yair100 you should find a more menu on the homepage next to the drives header ("...")
Currently it will open the windows dialog for mapping a network drive (probably behind the Files window!)

**Comment by yair100** [comment]
> @yair100 you should find a more menu on the homepage next to the drives header ("...")
> Currently it will open the windows dialog for mapping a network drive (probably behind the Files window!)

I see it now, and yeah, that bug with the window opening behind Files is getting to be pretty bad.

File: Files/UserControls/Widgets/DrivesWidget.xaml [hunk 1]
```diff
@@ -13,15 +13,55 @@
...
+                       FontSize="14"
+                       FontWeight="Medium"
+                       Opacity=".8"
+                       Text="Drives" />
 
+            <!--  Map Drive button  -->
+            <Button x:Uid="DrivesWidgetMoreOptionsButton"
+                    Grid.Column="1"
+                    Width="29"
+                    Height="25"
```
Comment (yaichenbaum) [inline Files/UserControls/Widgets/DrivesWidget.xaml:34]:
```suggestion
                    Width="36"
                    Height="32"
```

File: Files/UserControls/Widgets/DrivesWidget.xaml [hunk 2]
```diff
@@ -13,15 +13,55 @@
...
+                    Padding="0"
+                    HorizontalAlignment="Right"
+                    VerticalAlignment="Center"
+                    HorizontalContentAlignment="Center"
+                    VerticalContentAlignment="Center"
+                    Background="Transparent"
+                    Style="{StaticResource ButtonRevealStyle}"
+                    ToolTipService.ToolTip="More options...">
```
Comment (yaichenbaum) [inline Files/UserControls/Widgets/DrivesWidget.xaml:42]: Being that there is no text here, this button probably needs AutomationProperties.

**Review by yair100 — CHANGES_REQUESTED** [review]

**Comment by gave92** [comment]
Changes done! Regarding the issue of the dialog opening behind Files I'm not too worried as I'm planning to ship our own dialog in a future PR.

**Review by yair100 — APPROVED** [review]
Great work! And thank you for fixing all those random issues.

 :shipit:

**Pull request merged**